### ISSUE TITLE
REST API support?

### ISSUE THREAD

**Issue by HyShai** [notandy/ympd#63]
Is there a way to communicate with the ympd server using regular HTTP requests? e.g. I want to send `MPD_API_SET_PAUSE` without needing to load the ympd frontend - so I can quickly play/pause.
e.g. `http://raspberrypi.local:2938/?command=MPD_API_SET_PAUSE`

**Comment by kurt-vd** [comment]
Can't you just use a direct MPD connection?

**Comment by ajs124** [comment]
Yeah, I mean this is up for @notandy to decide, but IMHO this is not in the scope for this project.

If you really need a REST API for mpd for some reason and can't connect directly to it, you might want to implement this as a separate project.

**Comment by HyShai** [comment]
I figured that it would be a bit more trivial to piggyback on ympd's existing web socket support than writing a REST API from scratch. 

@kurt-vd It would be useful for mobile devices that don't have native MPD capabilities. E.g. iOS devices require an MPD app to talk mpd. Using ympd to send a simple pause/play command is a bit painful because it requires the entire web app to load.

**Comment by kurt-vd** [comment]
Hey,

> @kurt-vd It would be useful for mobile devices that don't have native MPD capabilities.

So you're looking at devices without dedicated MPD app, and want to control MPD via HTTP.

> E.g. iOS devices require an MPD app to talk mpd. Using ympd to send a simple pause/play
> command is a bit painful because it requires the entire web app to load.  

Are you looking for a (simple) CGI script that shows you some buttons, which is easier to load,
and that controls MPD from simple HTTP requests?
Without browsing the music collection and play queue, that looks 'trivial' to write.

The moment that you want feedback into the browser, ympd's websocket is the better alternative.
I understand you're not looking for feedback into the browser.

Kurt

**Comment by notandy** [comment]
Technically, it is very easy to parse all the websocket webclient -> ympd commands via http-post instead of websockets. But as @kurt-vd said, you will missing the websocket callbacks. For now websockets are the only way to communicate with ympd, but I could imagine that ympd implements some fallback rest/ajax support for browser with problems handling websockets.

**Comment by 1liminal1** [comment]
@HyShai did you ever find something that worked like this? Im looking to control MPD connections to Airplay devices via HA-BRIDGE? Basically use an echo dot to do multi room audio.

**Comment by kurt-vd** [comment]
I'm working on an mpd to mqtt bridge.
This is not HTTP rest, but works for me to integrate mpd control in my home automation

**Comment by heiderich** [comment]
> Technically, it is very easy to parse all the websocket webclient -> ympd commands via http-post instead of websockets. But as @kurt-vd said, you will missing the websocket callbacks. For now websockets are the only way to communicate with ympd, but I could imagine that ympd implements some fallback rest/ajax support for browser with problems handling websockets.

I would also welcome such a feature and I would not need any kind of feedback to the client.
